### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by k-malongo**
Done with signup

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -9,15 +9,15 @@ def create
...
-    def show
-        user = User.find_by(id: session[:user_id])
-        render json: user
+
+    def index
+        user = User.all
+        render json: user, status: :ok
     end
-    private
+
```
Comment (chibukasia) [inline app/controllers/users_controller.rb:20]: Add the show, update, and destroy actions for a user

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class UsersController < ApplicationController
-    before_action :authorize, only: [:show]
+
     def create
```
Comment (chibukasia) [inline app/controllers/users_controller.rb:5]: There is a validation method in the application controller that caters for all validations. Use the appropriate create and update

**Review by chibukasia — CHANGES_REQUESTED** [review]
Also work on the user serializers

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (k-malongo) [reply app/controllers/users_controller.rb:20]: Done the update and destroy

Comment (chibukasia) [reply app/controllers/users_controller.rb:20]: Okay

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (k-malongo) [reply app/controllers/users_controller.rb:5]: I have seen it

**Review by chibukasia — APPROVED** [review]

**Pull request merged**